docs/docs/ref-02-component-api.md

```diff
@@ -16,7 +16,7 @@ Instances of a React Component are created internally in React when rendering. T
 ```javascript
 setState(function|object nextState[, function callback])
 ```
-Merges nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
+Merges (shallow merge) nextState with the current state. This is the primary method you use to trigger UI updates from event handlers and server request callbacks.
```

Comment:
Maybe this?

"Performs a shallow merge of `nextState` into the current state."